scripts/BackupRetentionScript/RetentionScript.ps1

```diff
@@ -0,0 +1,194 @@
+    Thumbprint of the client certificate
+#>
+
+[CmdletBinding(PositionalBinding = $false)]
+param (
+    [Parameter(Mandatory=$false)]
+    [String] $UserName,
+
+    [Parameter(Mandatory=$false)]
+    [String] $FileSharePath,
+
+    [Parameter(Mandatory=$true)]
+    [String] $StorageType,
+
+    [Parameter(Mandatory=$true)]
+    [String] $DateTimeBefore,
+    
+    [Parameter(Mandatory=$false)]
+    [String] $ConnectionString,
+    
+    [Parameter(Mandatory=$false)]
+    [SecureString] $Password,
+
+    [Parameter(Mandatory=$false)]
+    [String] $ContainerName,
+
+    [Parameter(Mandatory=$false)]
+    [string] $StorageAccountName,
+
+    [Parameter(Mandatory=$false)]
+    [String] $StorageAccountKey,
+
+    [Parameter(Mandatory=$true)]
+    [String] $ClusterEndPoint,
+
+    [Parameter(Mandatory=$false)]
+    [Switch] $DeleteNotFoundPartitions,
+
+    [Parameter(Mandatory=$false)]
+    [String] $PartitionId,
+
+    [Parameter(Mandatory=$false)]
+    [String] $ServiceId,
+ 
+    [Parameter(Mandatory=$false)]
+    [String] $ApplicationId,
+
+    [Parameter(Mandatory=$false)]
+    [String] $ClientCertificateThumbprint
+)
+
+$command = ""
+if($StorageType -eq "FileShare")
+{
+  if(!$FileSharePath)
+  {
+    $FileSharePath = Read-Host -Prompt "Please enter the FileShare path"
+  }
+
+  if($UserName)
+  {
+      $command = $command +  ".\RetentionScriptFileShare.ps1 -UserName `"$UserName`" -FileSharePath `"$FileSharePath`" -DateTimeBefore `"$DateTimeBefore`" -ClusterEndPoint `"$ClusterEndPoint`""
+    if(!$Password)
+    {
+        $Password = Read-Host -Prompt "Please enter password for the userName: $UserName" -AsSecureString
+    }
+    $Global:Pass = $Password
+  }
+  else {
+    $command = $command +  ".\RetentionScriptFileShare.ps1 -FileSharePath `"$FileSharePath`" -DateTimeBefore `"$DateTimeBefore`" -ClusterEndPoint `"$ClusterEndPoint`""
+  }
+}
+elseif($StorageType -eq "AzureBlob")
+{
+    if($ConnectionString)
+    {
+        if($ContainerName)
+        {
+            $command = $command + ".\RetentionScriptAzureShare.ps1 -ConnectionString `"$ConnectionString`" -DateTimeBefore `"$DateTimeBefore`" -ClusterEndPoint `"$ClusterEndPoint`""
+        }
+        else {
+            $command = $command + ".\RetentionScriptAzureShare.ps1 -ConnectionString `"$ConnectionString`" -DateTimeBefore `"$DateTimeBefore`" -ClusterEndPoint `"$ClusterEndPoint`""
+        }
+    }
+    else {
+        if(!$StorageAccountName)
+        {
+            $StorageAccountName = Read-Host -Prompt "Please enter the Storage account name"
+        }
+        if(!$StorageAccountKey)
+        {
+            $StorageAccountKey = Read-Host -Prompt "Please enter the Storage account key"
+        }
+        $command = $command + ".\RetentionScriptAzureShare.ps1 -StorageAccountName `"$StorageAccountName`" -StorageAccountKey `"$StorageAccountKey`" -DateTimeBefore `"$DateTimeBefore`" -ContainerName `"$ContainerName`" -ClusterEndPoint `"$ClusterEndPoint`""    
```

Comment:
>RetentionScriptAzureShare [](start = 33, length = 25)

Rename! it is not AzureShare, replace this with AzureBlobStorage

Reply:
can i leave this also? I dont want to test the script again.